ister.py

```diff
@@ -1386,7 +1386,31 @@ def validate_template(template):
     LOG.debug(template)
 
 
-def process_kernel_cmdline(f_kcmdline, sleep_time=15):
+def download_ister_conf(uri, timeout=15):
+    """Download the ister.conf/ister.json file from 'uri' to a local temporary
+    file and return the temporary file path. The timeout argument specifies for
+    how long to try downloading the file."""
+
+    tmpfd, abs_path = tempfile.mkstemp()
+    LOG.debug("ister_conf tmp file = {0}".format(abs_path))
+
+    start_time = time.time()
+    while True:
+        try:
+            with request.urlopen(uri) as response:
+                with closing(os.fdopen(tmpfd, "wb")) as out_file:
+                    shutil.copyfileobj(response, out_file)
+                    return abs_path
+            os.unlink(abs_path)
```

Comment:
I think the previous version had this unlink removed because it couldn't be executed (either exception or success). Was there a reason for adding this back or am I confused because of my bad merging?